backend/src/auth/service.py

```diff
@@ -21,21 +24,88 @@ def validate_user_credentials(db_session: Session, username: str, password: str)
+        username: str = payload.get("sub")
+        if username is None:
+            logger.warning("Token missing 'sub' field")
+            raise HTTPException(
+                status_code=status.HTTP_401_UNAUTHORIZED,
+                detail="Invalid token: missing subject",
+                headers={"WWW-Authenticate": "Bearer"},
+            )
+        
+        # 查詢用戶
+        user = db.query(User).filter(User.username == username).first()
+        if user is None:
+            logger.warning(f"User not found for token: {username}")
+            raise HTTPException(
+                status_code=status.HTTP_404_NOT_FOUND,
+                detail="User not found",
+            )
+        logger.info(f"User {username} authenticated successfully with token")
+        return user
+    
+    except ExpiredSignatureError as e:
+        capture_exception(e)
+        logger.error(f"Token has expired: {e}")
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="Token has expired",
+        )
+
+    except JWTError as e:
+        capture_exception(e)
+        logger.error(f"Invalid token error: {e}")
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="Invalid token",
+            headers={"WWW-Authenticate": "Bearer"},
+        )
+
+    except Exception as e:
+        capture_exception(e)
+        logger.error(f"Unexpected error during token authentication: {str(e)}")
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail="An unexpected error occurred",
+        )
 
 def create_access_token(user_data, expires_delta=None):
-    to_encode = user_data.copy()
-    if expires_delta:
-        expire = datetime.utcnow() + expires_delta
-    else:
-        expire = datetime.utcnow() + timedelta(minutes=15)
-    to_encode.update({"exp": expire})
-    print(to_encode)
-    encoded_jwt = jwt.encode(to_encode, auth_config.SECRET_KEY, algorithm=auth_config.ALGORITHM)
-    return encoded_jwt
\ No newline at end of file
+    try:
+        to_encode = user_data.copy()
+        
+        # 設定過期時間
+        if expires_delta:
+            expire = datetime.utcnow() + expires_delta
+        else:
+            expire = datetime.utcnow() + timedelta(minutes=15)
+        
+        to_encode.update({"exp": expire})
+        
+        # 編碼 JWT
+        encoded_jwt = jwt.encode(to_encode, "your_secret_key", algorithm="HS256")
+        logger.info(f"Access token created for user: {user_data['sub']}")
+        return encoded_jwt
+    
+    except AttributeError as e:
+        capture_exception(e)
+        logger.error(f"Attribute error while creating access token: {str(e)}")
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail="An internal error occurred while creating the access token",
```

Comment:
Messages returned to client can be not that detailed, as that is not necessary for client.